### PR TITLE
feat: add wallet functionality

### PR DESCRIPTION
**Pull request by edgarkhanzadian**
https://github.com/user-attachments/assets/ffdcc52e-1320-4fba-b4d5-2aac54e29156

### CODE REVIEW

File: packages/crypto/src/keychain.ts [hunk 1]
```diff
@@ -32,8 +32,8 @@ export function deriveKeychainFromXpub(xpub: string) {
  * Gets keychain fingerprint directly from mnemonic. This is useful for
  * referencing a mnemonic safely by an identifier.
  */
-export async function getMnemonicRootKeyFingerprint(mnemonic: string) {
-  const keychain = deriveRootBip32Keychain(await deriveBip39SeedFromMnemonic(mnemonic));
+export async function getMnemonicRootKeyFingerprint(mnemonic: string, passphrase?: string) {
+  const keychain = deriveRootBip32Keychain(await deriveBip39SeedFromMnemonic(mnemonic, passphrase));
```
Comment (kyranjamie) [inline packages/crypto/src/keychain.ts:36]: Can you add some tests for this?

File: apps/mobile/src/state/storage-persistors.ts [hunk 2]
```diff
@@ -43,9 +88,10 @@ export function mnemonicStore(fingerprint: string) {
       }
       return mnemonicSchema.parse({ mnemonic });
     },
-    async setMnemonic(mnemonic: string, passphrase?: string) {
-      if (passphrase) await SecureStore.setItemAsync(passphraseKey, passphrase, secureStoreConfig);
-      return SecureStore.setItemAsync(fingerprint, mnemonic, secureStoreConfig);
+    async setMnemonic(mnemonic: string, biometrics: boolean, passphrase?: string) {
```
Comment (kyranjamie) [inline apps/mobile/src/state/storage-persistors.ts:91]: Passing in booleans can make reading the [fn signatures unclear](https://github.com/ryanmcdermott/clean-code-javascript?tab=readme-ov-file#dont-use-flags-as-function-parameters)

`setMnemonic(mnemonic, passphrase, { biometrics: boolean }` is much more readable

Why is it we need it configurable again?

File: apps/mobile/src/app/wallet/create-new-wallet.tsx [hunk 3]
```diff
@@ -10,27 +10,43 @@ import { Button } from '@/components/button';
...
   const router = useRouter();
-  const test_array = getTestMnemonic().split(' ');
   const [isHidden, setIsHidden] = useState(true);
   const [isCopied, setIsCopied] = useState(false);
+  const [mnemonic, setMnemonic] = useState<string | null>(null);
+
+  useEffect(() => {
+    const tempMnemonic = generateMnemonic();
+    setMnemonic(tempMnemonic);
+    tempMnemonicStore.setTemporaryMnemonic(tempMnemonic);
```
Comment (kyranjamie) [inline apps/mobile/src/app/wallet/create-new-wallet.tsx:47]: I can wire this up to store separately

File: apps/mobile/src/app/wallet/create-new-wallet.tsx [hunk 4]
```diff
@@ -10,27 +10,43 @@ import { Button } from '@/components/button';
...
 
+import { generateMnemonic } from '@leather.io/crypto';
 import { Box, Text, Theme, TouchableOpacity } from '@leather.io/ui/native';
 
-function getTestMnemonic() {
-  if (process.env.EXPO_PUBLIC_SECRET_KEY) {
-    return process.env.EXPO_PUBLIC_SECRET_KEY;
+function MnemonicDisplay({ mnemonic }: { mnemonic: string | null }) {
+  if (!mnemonic) {
+    return null;
```
Comment (fbwoolf) [inline apps/mobile/src/app/wallet/create-new-wallet.tsx:24]: return on same line?

File: apps/mobile/src/app/wallet/create-new-wallet.tsx [hunk 5]
```diff
@@ -10,27 +10,43 @@ import { Button } from '@/components/button';
...
   }
-  throw new Error('No mnemonic is provided in EXPO_PUBLIC_SECRET_KEY');
+  const mnemonicArr = mnemonic.split(' ');
```
Comment (fbwoolf) [inline apps/mobile/src/app/wallet/create-new-wallet.tsx:26]: Wondering abt using `Arr` in naming, maybe something more descriptive would be better? We know it is an array from the method. `mnemoicWords` or something? Can ignore comment.

**Review by fbwoolf — APPROVED** [review]

File: apps/mobile/src/app/wallet/create-new-wallet.tsx (hunk 3 above)
Comment (kyranjamie) [reply apps/mobile/src/app/wallet/create-new-wallet.tsx:47]: Yeah makes sense, just need to make sure this is the mnemonic that gets added to the store

File: apps/mobile/src/app/wallet/create-new-wallet.tsx (hunk 5 above)
Comment (kyranjamie) [reply apps/mobile/src/app/wallet/create-new-wallet.tsx:26]: +1 to favouring plural or something like MnemonicWordList to avoid naming as its js type

**Pull request merged**